tests/integration/messages/create_messages/test_field_validation.py

```diff
@@ -312,28 +273,187 @@ def test_null_personalisation(bearer_token_int, correlation_id, personalisation)
+        },
+        json=data
+    )
 
     Assertions.assert_error_with_optional_correlation_id(
         resp,
         400,
         Generators.generate_null_value_error("/data/attributes/personalisation"),
         correlation_id
     )
+
+
+@pytest.mark.inttest
+@pytest.mark.parametrize("correlation_id", CORRELATION_IDS)
+def test_invalid_sms_contact_details(bearer_token_int, correlation_id):
+    """
+    .. include:: ../../partials/validation/test_invalid_contact_details_sms.rst
+    """
+    data = Generators.generate_valid_create_message_body("int")
+    data["data"]["attributes"]["recipient"]["contactDetails"] = {"sms": "07700900002"}
+    resp = requests.post(
+        f"{INT_URL}{MESSAGES_ENDPOINT}",
+        headers={
+            "Authorization": bearer_token_int.value,
+            **headers,
+            "X-Correlation-Id": correlation_id
+        },
+        json=data
+    )
+
+    Assertions.assert_error_with_optional_correlation_id(
+        resp,
+        400,
+        Generators.generate_invalid_value_error_custom_detail(
+            "/data/attributes/recipient/contactDetails/sms",
+            "Input failed format check"
+        ),
+        correlation_id
+    )
+
+
+@pytest.mark.inttest
+@pytest.mark.parametrize("correlation_id", CORRELATION_IDS)
+def test_invalid_email_contact_details(bearer_token_int, correlation_id):
+    """
+    .. include:: ../../partials/validation/test_invalid_contact_details_email.rst
+    """
+    data = Generators.generate_valid_create_message_body("int")
+    data["data"]["attributes"]["recipient"]["contactDetails"] = {"email": "invalidEmailAddress"}
+    resp = requests.post(
+        f"{INT_URL}{MESSAGES_ENDPOINT}",
+        headers={
+            "Authorization": bearer_token_int.value,
+            **headers,
+            "X-Correlation-Id": correlation_id
+        },
+        json=data
+    )
+
+    Assertions.assert_error_with_optional_correlation_id(
+        resp,
+        400,
+        Generators.generate_invalid_value_error_custom_detail(
+            "/data/attributes/recipient/contactDetails/email",
+            "Input failed format check"
+        ),
+        correlation_id
+    )
+
+
+@pytest.mark.inttest
+@pytest.mark.parametrize("correlation_id", CORRELATION_IDS)
+def test_invalid_address_contact_details_too_few_lines(bearer_token_int, correlation_id):
+    """
+    .. include:: ../../partials/validation/test_invalid_contact_details_address_lines_too_few.rst
+    """
+    data = Generators.generate_valid_create_message_body("int")
+    data["data"]["attributes"]["recipient"]["contactDetails"] = {
+        "address": {
+            "lines": ["1"],
+            "postcode": "test"
+        }
+    }
+    resp = requests.post(
+        f"{INT_URL}{MESSAGES_ENDPOINT}",
+        headers={
+            "Authorization": bearer_token_int.value,
+            **headers,
+            "X-Correlation-Id": correlation_id
+        },
+        json=data
+    )
+
+    error = Error(
```

Comment:
Is there a reason this is declared in the file rather than in constants?

Reply:
Just because I'm doing my best to follow the existing code style, [this is copied from here](https://github.com/NHSDigital/communications-manager-api/blob/fdb2e14bde13133e5008ba2fca268c4357fc163c/tests/development/messages/create_messages/test_field_validation.py#L376)

Would you like them both changed to use constants or to stay as they are? I like this way because it decouples the tests for different environments as we previously discussed.

Reply:
Implemented